geocity/apps/submissions/forms.py

```diff
@@ -703,15 +721,43 @@ def get_list_multiple_field_kwargs(self, field, default_kwargs):
             else forms.CheckboxSelectMultiple(),
         }
 
+    def get_geom_field_kwargs(self, field, default_kwargs):
+
+        options = {
+            "map_widget_configuration": [field.map_widget_configuration.configuration],
+        }
+
+        widget = GeometryWidgetAdvanced(attrs={"options": options})
+        widget.attrs["options"]["edit_geom"] = True
+
+        return {
+            **default_kwargs,
+            "widget": widget,
+        }
+
     def save(self):
         to_geocode_addresses = []
         for form, field in self.get_fields():
-            if field.is_value_field():
+            if (
+                field.is_value_field()
+                and not field.input_type == models.Field.INPUT_TYPE_GEOM
+            ):
                 self.instance.set_field_value(
                     form=form,
                     field=field,
                     value=self.cleaned_data[self.get_field_name(form, field)],
                 )
+            if (
+                field.input_type == models.Field.INPUT_TYPE_GEOM
+                and self.cleaned_data[self.get_field_name(form, field)]
+            ):
```

Comment:
This case should never happen, but it's better to keep checking if field.is_value_field().
It could possibly return something like "input_type doesn't exist on field"
`
if (
                field.is_value_field()
                and field.input_type == models.Field.INPUT_TYPE_GEOM
                and self.cleaned_data[self.get_field_name(form, field)]
            ):
`